src/main/java/org/openhab/tools/analysis/checkstyle/PomXmlCheck.java

```diff
@@ -137,6 +135,10 @@ private void processManifestFile(FileText fileText) throws CheckstyleException {
         // We need this in order to filter the "qualifier" for the Snapshot versions
         manifestVersion = getVersion(version.toString(), manifestVersionPattern);
 
+        if (manifestVersion == null) {
+            log(0, String.format(INCORRECT_VERSION, manifestVersionRegularExpression));

```

Comment:
OK, but what if the manifest version is correct and the pom version is not ? I would expect a similar message (e.g. `The pom.xml version did not match the requirements`).

Please add both test cases:
- pom.xml version does not match, but manifest version matches
- manifest version does not match, but pom version matches

Then it will be visible what message is logged in each case.